x-pack/plugins/index_management/public/hooks/use_details_page_mappings_model_management.ts

```diff
@@ -27,20 +28,24 @@ const getCustomInferenceIdMap = (
 ) => {
   return models?.data.reduce<InferenceToModelIdMap>((inferenceMap, model) => {
     const inferenceId = model.model_id;
-    const trainedModelId =
-      'model_id' in model.service_settings ? model.service_settings.model_id : '';
 
+    const trainedModelId =
+      'model_id' in model.service_settings &&
+      (model.service_settings.model_id === ElasticsearchModelDefaultOptions.elser ||
+        model.service_settings.model_id === ElasticsearchModelDefaultOptions.e5)
+        ? model.service_settings.model_id
+        : '';
     inferenceMap[inferenceId] = {
       trainedModelId,
-      isDeployable: model.service === 'elser' || model.service === 'elasticsearch',
+      isDeployable: model.service === Service.elser || model.service === Service.elasticsearch,
       isDeployed: deploymentStatsByModelId[trainedModelId] === 'deployed',
```

Comment:
Is it possible to replace the hard-coded 'deployed' with a constant just like this [change](https://github.com/elastic/kibana/pull/183869/files#diff-1e216f4b255e2eff69f49f9ad31f95408383e8f23b07c76b37f024a0194ef467R40)?

Reply:
We should have a constant, I will add in next PR as I will be modifying the deployment toasts a bit